js-miniapp-bridge/src/common-bridge.ts

```diff
@@ -121,9 +121,25 @@ export class MiniAppBridge {
    * mini app that listen to this eventType.
    * @param  {[String]} eventType EventType which will be used to listen for the event
    * @param  {[String]} value Additional message sent from the native on invoking for the eventType
+   * @param  {[String]} navigationBarHeight Additional message sent from the native on invoking for the navigationBarHeight
+   * @param  {[String]} screenHeight Additional message sent from the native on invoking for the screenHeight
+   * @param  {[String]} keyboardHeight Additional message sent from the native on invoking for the keyboardHeight
    */
-  execCustomEventsCallback(eventType: string, value: string) {
-    const event = new CustomEvent(eventType, { detail: value });
+  execCustomEventsCallback(
+    eventType: string,
+    value: string,
+    navigationBarHeight: number,
+    screenHeight: number,
+    keyboardHeight: number
+  ) {
+    const event = new CustomEvent(eventType, {
+      detail: {
+        message: value,
+        navigationBarHeight,
+        screenHeight,
+        keyboardHeight,
```

Comment:
Isn't this function and `CustomEvent` also used for the `resume` and `pause` events? So this would mean that those events always contain `height` properties for all of the events but they're `null` in the case of `pause`, `resume`, etc.

I think it would be more clear to the SDK users if we have a specific type for the keyboard events, i.e. `MiniAppKeyboardEvent` or something. I'm not sure the best way to handle this, but maybe we need a separate function for `execKeyboardEvent`?

Reply:
@corycaywood I fixed it just I logged the `execCustomEventsCallback` & `execKeyboardEventsCallback` and saw that the logs duplicates with every call is that something we have to take a deeper look into?

Reply:
@laubengaier Do you mean that the callback for `execCustomEventsCallback` was getting called twice even before the keyboard callback was added?

It does seem like it would be a problem, because the mini app's callback would be invoked twice.

Reply:
@laubengaier where did you spot this log? on iOS console in Sample App project?

Reply:
@corycaywood @Climbatize 

I've already added an extra queue for the keyboards events but it seems to trigger it js side a bunch of times. I've changed the bridge code inside the iOS project to log the keyboard event like below:

```
     MiniAppBridge.prototype.execKeyboardEventsCallback = function (eventType, message, navigationBarHeight, screenHeight, keyboardHeight) {
         console.log('### keyboard event: ' + eventType);
         var event = new CustomEvent(eventType, {
             detail: {
                 message: message,
                 navigationBarHeight: navigationBarHeight,
                 screenHeight: screenHeight,
                 keyboardHeight: keyboardHeight,
             },
         });
         var queueObj = mabKeyboardEventQueue.filter(function (customEvent) { return customEvent === event; })[0];
         if (!queueObj) {
             
             if (eventType === event.type) {
                 console.log('### keyboard remove event: ' + event.type);
                 removeFromKeyboardEventQueue(event);
             }
             queueObj = event;
             mabKeyboardEventQueue.unshift(queueObj);
         }
         this.executor.execEvents(queueObj);
     };
```
ends up looking like this:
![Screen Shot 2022-03-14 at 12 10 58](https://user-images.githubusercontent.com/17570451/158099289-6fc391d4-93e9-4ed3-8463-554dc35b51e1.png)

not too sure about queues in js

Reply:
@laubengaier Seems like it must be an issue with the JS Sample App. Actually, `removeEventListener` is never called by the Sample App, so it is probably being attached multiple times without removing the old one. Probably we should hook into the React component state to add/remove the listeners.

I think we can address that later in another PR, so this one is okay to merge for now.